libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -518,7 +518,25 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 				arguments[i]->accept(*this);
 				utils().convertType(*arguments[i]->annotation().type, *function.parameterTypes()[i]);
 			}
-			_functionCall.expression().accept(*this);
+
+			{
+				bool shortcutTaken = false;
+				if (auto identifier = dynamic_cast<Identifier const*>(&_functionCall.expression()))
+					if (auto functionDef = dynamic_cast<FunctionDefinition const*>(identifier->annotation().referencedDeclaration))
+					{
+						// Do not directly visit the identifier, because this way, we can avoid
+						// the runtime entry label to be created at the creation time context.
+						CompilerContext::LocationSetter locationSetter2(m_context, *identifier);
+						m_context << m_context.functionEntryLabel(m_context.resolveVirtualFunction(*functionDef)).pushTag();
+						if (m_context.runtimeContext())
```

Comment:
This is needed in the case it is used in both deploy and runtime?

Reply:
It is needed because internal function pointers are unpacked differently depending on whether we are at deploy time or at runtime.

Reply:
Yeah but this `if` is only here because it may be used in both.

I think it may make sense extending `pushCombinedFunctionEntryLabel` with a flag `runtimeOnly` (if false it won't include the runtime label) and it that case we can use it here.

Reply:
Yes, seems cleaner.